python_modules/dagster/dagster/_core/definitions/auto_materialize_condition.py

```diff
@@ -56,9 +59,15 @@ class MissingAutoMaterializeCondition(NamedTuple):
 @whitelist_for_serdes
 class ParentOutdatedAutoMaterializeCondition(NamedTuple):
     """Indicates that this asset should be skipped because one or more of its parents are outdated.
+    This could be because the parent isn't going to run, or it could be that the parent is going to be materialized
+    but in a separate run due to different partitions or repositories.
     """
 
     decision_type: AutoMaterializeDecisionType = AutoMaterializeDecisionType.SKIP
+    parent_asset_key: Optional["AssetKey"] = None
+    parent_will_materialize: Optional[bool] = None
+    different_partitions: Optional[bool] = None
```

Comment:
From the name, I expected the type of this would be a collection. Something like `parent_has_different_partitions_def` would be a clearer name,